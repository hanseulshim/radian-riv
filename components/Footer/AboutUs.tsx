import React, { useState, useCallback } from 'react'
import Modal from 'components/common/Modal'
import { getAboutUs } from 'api'
import Table from 'components/common/Table'

interface Props {
  closeModal: () => void
}

export default function AboutUs({ closeModal }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const tableData = await getAboutUs()
    if (tableData.length) {
      const tableColumns = Object.keys(tableData[0]).map(key => ({
        Header: key,
        accessor: key
      }))
      setColumns(tableColumns)
    }
    setData(tableData)
  }, [])

  const getColWidth = (header: string) => ({
    minWidth: 250
  })

  return (
    <Modal closeModal={closeModal} title={'About Us'} percent={80}>
      <Table
        columns={columns}
        data={data}
        fetchData={fetchData}
        getColWidth={getColWidth}
      />
    </Modal>
  )
}
