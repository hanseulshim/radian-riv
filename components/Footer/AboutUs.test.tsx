import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import AboutUs from './AboutUs'

describe('About Us', () => {
  let component
  beforeAll(() => {
    component = shallow(<AboutUs closeModal={() => {}} />)
  })
  it('should have one child', () => {
    expect(component.children()).toHaveLength(1)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
