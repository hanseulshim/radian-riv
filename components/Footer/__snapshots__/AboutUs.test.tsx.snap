// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us should match snapshot 1`] = `
<Modal
  closeModal={[Function]}
  percent={80}
  title="About Us"
>
  <Table
    columns={Array []}
    data={Array []}
    fetchData={[Function]}
  />
</Modal>
`;
